code/_onclick/other_mobs.dm

```diff
@@ -297,6 +309,26 @@
 	else
 		AttackingTarget(attack_target)
 
+/*
+	Lightgeists (those little healers from xenobiology and the colossus crystal)
+*/
+
+/mob/living/simple_animal/hostile/lightgeist/resolve_unarmed_attack(atom/attack_target, list/modifiers)
+	attack_target.attack_lightgeist(src, modifiers)
+
+/mob/living/silicon/hostile/lightgeist/resolve_right_click_attack(atom/target, list/modifiers)
+	return target.attack_lightgeist_secondary(src, modifiers)
+
+/atom/proc/attack_lightgeist(mob/living/user, list/modifiers)
+	return
```

Comment:
Alas, it's the most direct way to deal with the issue I can think of right now. Ever since I had to refactor unarmed attack code, it's just way easier to add more attack_x procs to the game. 😔